usaspending_api/common/threaded_data_loader.py

```diff
@@ -52,117 +57,129 @@ def __init__(self, model_class, threads=10, field_map={}, value_map={}, collisio
     def load_from_file(self, filepath):
         self.logger.info('Started processing file ' + filepath)
         # Create the Queue object - this will hold all the rows in the CSV
-        row_queue = queue.Queue(maxsize=100)
-        self.exit_flag = False
-
-        # Spawn our threads
+        row_queue = JoinableQueue()
+
+        references = {
+            "collision_field": self.collision_field,
+            "collision_behavior": self.collision_behavior,
+            "logger": self.logger,
+            "pre_row_function": self.pre_row_function,
+            "post_row_function": self.post_row_function,
+            "fields": self.fields.copy()
+        }
+
+        # Spawn our processes
+        db.connections.close_all()
         pool = []
-        while len(pool) < self.threads:
-            thread = DataLoaderThread("Thread-" + str(len(pool)), row_queue, self)
-            thread.start()
-            pool.append(thread)
+        while len(pool) < self.processes:
+            process = DataLoaderThread("Process-" + str(len(pool)), self.model_class, row_queue, self.field_map.copy(), self.value_map.copy(), references)
+            pool.append(process)
```

Comment:
Small detail, but this might be a wee bit simpler:
```python
pool = [DataLoaderThread('Process-' + str(i), self.model_class, row_queue, ...)
         for i in range(self.processes)]
```